tests/ShadowDOM/js/test.js

```diff
@@ -520,4 +524,40 @@ suite('Shadow DOM', function() {
 
   });
 
+  // https://github.com/webcomponents/webcomponentsjs/issues/316
+  test('nested renderer should invalidate attributes', function() {
+    var div = document.createElement('div');
+    div.innerHTML =
+        '<x-outer></x-outer>' +
+        '<template id="x-host">' +
+          'Foo: [<content select="[foo]"></content>] ' +
+          'Bar: [<content select="[bar]"></content>]' +
+        '</template>' +
+        '<template id="x-outer">' +
+          '<x-host id="host">' +
+            '<child foo>~Foo~</child> ' +
+            '<child foo bar>~Foo+Bar~</child> ' +
```

Comment:
Sure. But to be clear, this comment is not related to this issue. The white space problem is a different bug: I reproduced it without the fix. I added the comment here on the pull request, to tie back to my comment in investigation notes:
https://github.com/webcomponents/webcomponentsjs/issues/316#issuecomment-106635479

But anyway, happy to add a comment to the code if it means landing this :)